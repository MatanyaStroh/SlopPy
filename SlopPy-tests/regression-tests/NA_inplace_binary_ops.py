# do a bunch of in-place binary operations on an NA object

x = 1 / 0
assert type(x) is NA

x += 5
x -= 5
x *= 5
x /= 5
x //= 5
x %= 5
x <<= 5
x >>= 5
x &= 5
x ^= 5
x |= 5


y = 10
y += x
assert y is x
y = 10
y -= x
assert y is x
y = 10
y *= x
assert y is x
y = 10
y /= x
assert y is x
y = 10
y //= x
assert y is x
y = 10
y %= x
assert y is x
y = 10
y <<= x
assert y is x
y = 10
y >>= x
assert y is x
y = 10
y &= x
assert y is x
y = 10
y ^= x
assert y is x
y = 10
y |= x
assert y is x

