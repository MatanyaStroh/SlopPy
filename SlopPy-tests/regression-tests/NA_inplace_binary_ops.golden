{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/NA_inplace_binary_ops.py", line 3, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/NA_inplace_binary_ops.py', '__doc__': None, '__package__': None}
}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:6}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:7}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:8}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:9}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:10}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:11}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:12}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:13}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:14}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:15}
{binary_op(NA,*) | regression-tests/NA_inplace_binary_ops.py <module>:16}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:20}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:23}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:26}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:29}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:32}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:35}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:38}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:41}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:44}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:47}
{binary_op(*,NA) | regression-tests/NA_inplace_binary_ops.py <module>:50}
